### ISSUE TITLE
Obtaining the Azure automation reg key via ARM

### ISSUE THREAD

**Issue by mshorrosh** [azureautomation/automation-packs#7]
Hi

Is there a way to obtain the registration key for AA via ARM to pass to the VM DSC Extension instead of running powershell to deploy AA and query for reg key and pass that in as a parameter?

Thanks,
Matt

**Comment by JAL9302** [comment]
Hi,

Have you tried https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#listkeys?

**Comment by mshorrosh** [comment]
Hi,

I saw this but the documentation said it was for storage accounts, so I didn't think it would work for pulling Azure Automation's keys. I will give it a shot though later and report back if I have any luck.

Matt

**Comment by JAL9302** [comment]
I just checked and looks like this isn't possible right now. Will open a bug internally to fix this, in the same way listKeys works for storage accounts.

**Comment by mshorrosh** [comment]
Hi Joe

Is there any updates on obtaining this reg key via arm rgt functions?

Matt

**Comment by JAL9302** [comment]
Not yet. I just pinged our dev managers again to request we get this done.

**Comment by burn2delete** [comment]
+1

**Comment by burn2delete** [comment]
Is there somewhere else we should be tracking the status of this?

**Comment by JAL9302** [comment]
Yes, you can track the issue here: https://feedback.azure.com/forums/246290-automation/suggestions/12964470-provide-listkeys-functionality-through-the-api-si. Please upvote the request at that link if this is an important scenario for you so we can better prioritize it.

**Comment by dpurge** [comment]
No progress on this yet?
